streams/src/main/java/org/apache/kafka/streams/state/internals/Segment.java

```diff
@@ -17,12 +17,12 @@
 package org.apache.kafka.streams.state.internals;
 
 import org.apache.kafka.common.utils.Bytes;
-import org.apache.kafka.streams.processor.StateStore;
 import org.apache.kafka.streams.state.KeyValueIterator;
+import org.apache.kafka.streams.state.KeyValueStore;
 
 import java.io.IOException;
 
-public interface Segment extends StateStore {
+public interface Segment extends KeyValueStore<Bytes, byte[]>, BulkLoadingStore {
```

Comment:
We keep the extended super class intentionally so that in the future we can implement the segment differently (e.g. not based on rocksDB at all), but since this is an internal class anyways we can always change it back when needed to, so I guess this is fine --- just to leave the context on this.